src/main.rs

```diff
@@ -354,7 +359,10 @@ extern "C" fn kernel_init(hart_id: u64, fdt_addr: u64) {
         // unless we have support for per-mode filtering.
         println!("Sscofpmf support present");
         if let Err(e) = PmuInfo::init() {
+            test_fail!("sstc presence");
             println!("PmuInfo::init() failed with {:?}", e);
+        } else {
+            test_pass!("sstc presence")
         }
     }
```

Comment:
Added test_declare_pass!() and test_runtest!()
test_declare_pass!() declares and passes in one macro

test_runtest!() executes a block. There is also test_assert!()

Reply:
Can we convert the remaining instances of separate `test_declare!` + `test_pass!` pairs to a single invocation? Having both strings matches is brittle and I'd like to avoid checking it in...

Reply:
fixed